RD-1123 Added documentation for filters widget (#1795)
cloudify-cosmo/docs.getcloudify.org: content/working_with/console/widgets/filters.md

---
layout: bt_wiki
title: Filters
category: Widgets
draft: false
---

Displays the list of defined filters.
The table has 4 columns:

* **Filter name** - filter ID
* **Creator** - filter creator
* **Created** - filter creation date and time
* Actions column

![filters]( /images/ui/widgets/filters.png )

> Actions column contains following action icons:

Reviewer: ```suggestion
Actions column contains the following action icons:
```

> Actions column contains the following action icons:

Reviewer 2: ```suggestion
The actions column contains the following action icons:
```

> * **Edit filter** opening filter rules edit modal (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule specifying)

Reviewer 2: ```suggestion
* **Edit filter** opens the **filter rules edit** modal (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule definition)
```

> * **Clone filter** opening filter clone modal allowing to create a modified version of the selected filter (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule specifying)

Reviewer 2: ```suggestion
* **Clone filter** opens the **filter clone** modal allowing to create a modified copy of the selected filter (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule definition)
```

> * **Clone filter** opening filter clone modal allowing for creating a possibly modified version of selected filter (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule specifying)

Reviewer: Optional:

Probably a bit easier to read:
```suggestion
* **Clone filter** opening filter clone modal allowing to create a modified version of the selected filter (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule specifying)
```

> * **Delete filter** allowing for removal of selected filter (see note below)

Reviewer 2: ```suggestion
* **Delete filter** removes the selected filter (see note below)
```

> Deleting filter which is used as a default filter in Deployments View widget is not possible.

Reviewer 2: ```suggestion
Filters used as a default filter in the Deployments View widget cannot be deleted.
```

When trying to delete such a filter a modal shows up describing where (on which page and in which widget) and by whom (by which user) the filter is used.

![filters delete]( /images/ui/widgets/filters-delete.png ) 

> Clicking the `Add` button above the table opens the filter add modal allowing for creation of a new filter by specifying filter ID and filter rules.

Reviewer: Optional:

As this is a long sentence, maybe a bit easier to read would be:
```suggestion
Clicking the `Add` button above the table opens the filter add modal allowing to create a new filter by specifying filter ID and filter rules.
```
?

## Defining filter rules

Add, edit and clone operation modals share a common component for defining filter rules.

> ![filters rules]( /images/ui/widgets/filters-rules.png ) 

Reviewer: The styling for label rule will change probably once I implement RD-2007.

Author: Looks like the only important change will be the space between key and value dropdowns. If so I could manually edit the image and introduce spacing so the image reflected future change and the PR was not blocked.

Reviewer: In case of this screenshot, it seems the only change.

Alternatively, you can also generate it using the code from https://github.com/cloudify-cosmo/cloudify-stage/pull/1315 .

The component presents a list of rows, each representing a single filter rule. Each row contains three inputs:

> * Rule type selection dropdown allowing for selecting if the given rule applies to labels or one of supported attributes (like blueprint, site name or creator)

Reviewer: ```suggestion
* Rule type selection dropdown allowing for selecting if the given rule applies to labels or one of the supported attributes (like blueprint, site name or creator)
```

---
layout: bt_wiki
title: Filters
category: Widgets
draft: false
---

Displays the list of defined filters.
The table has 5 columns:

* **Filter name** - filter ID
* **Creator** - filter creator
* **Created** - filter creation date and time
* **System** - system filter indicator - checked when given filter is a system filter, unchecked otherwise
* Actions column

![filters]( /images/ui/widgets/filters.png )

The actions column contains the following action icons:

* **Clone filter** opens the **filter clone** modal allowing to create a modified copy of the selected filter (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule definition)
* **Edit filter** available only to `user` filters, opens the **filter rules edit** modal (see [defining filter rules]({{< relref "#defining-filter-rules" >}}) for details on rule definition)

> * **Delete filter** available only to `user` filters, removes the selected filter (see note below)

Reviewer 2: ```suggestion
* **Delete filter** available only to non-`system` filters, removes the selected filter (see note below)
```

Reviewer 2: we dont mention "users filter above. we do mention system filters so changing the phrasing to simplify the understanding.

Filters used as a default filter in the Deployments View widget cannot be deleted.
When trying to delete such a filter a modal shows up describing where (on which page and in which widget) and by whom (by which user) the filter is used.

![filters delete]( /images/ui/widgets/filters-delete.png ) 

Clicking the `Add` button above the table opens the filter add modal allowing to create a new filter by specifying filter ID and filter rules.

## Defining filter rules

Add, edit and clone operation modals share a common component for defining filter rules.

![filters rules]( /images/ui/widgets/filters-rules.png ) 

The component presents a list of rows, each representing a single filter rule. Each row contains three inputs:

> * Rule type selection dropdown allowing for selecting if the given rule applies to labels or one of the supported attributes (like blueprint, site name or creator)

Reviewer 2: ```suggestion
* Rule type selection dropdown - selecting the context of the rule which can be based on labels or supported deployment attributes such as blueprint ID, site name, or creator.
```
